src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRMultiConnectionInfoBinding/MultiConnectionInfoAsyncConverter.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.SignalR;
+using Microsoft.Azure.SignalR.Management;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    internal class MultiConnectionInfoAsyncConverter : IAsyncConverter<SignalRMultiConnectionInfoAttribute, Dictionary<ServiceEndpoint, SignalRConnectionInfo>>
+    {
+        private readonly IServiceManagerStore _serviceManagerStore;
+
+        public MultiConnectionInfoAsyncConverter(IServiceManagerStore serviceManagerStore)
+        {
+            _serviceManagerStore = serviceManagerStore;
+        }
+
+        public async Task<Dictionary<ServiceEndpoint, SignalRConnectionInfo>> ConvertAsync(
```

Comment:
cx get `Dictionary<ServiceEndpoint, SignalRConnectionInfo>`, so the router need to be outside of extension? why you don't use the endpoint router and then return a `SignalRConnectionInfo` so that cx can simply return it.

Reply:
oh, I see, it's mainly for js, right?

Reply:
yes